### PR TITLE
[flutter_tools] remove runFromSource, move runInView to vm_service extension

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Remove runFromSource, which is just a wrapper
Move runInView to the vmService extension
Remove unused reference to .packages file

## Related Issues

Work towards https://github.com/flutter/flutter/issues/52179

## Tests

I added a new test helper for dealing with the vmservice. It is based on the FakeProcessManager, but is a tad more complex due to streamNotify, essentially responses without a corresponding request.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
I need to stop accidentally mocking extension methods

File: packages/flutter_tools/test/general.shard/vmservice_test.dart [hunk 1]
```diff
@@ -280,6 +281,104 @@ void main() {
...
+      'method': 'streamNotify',
+      'params': <String, Object>{
+        'streamId': streamId,
+        'event': event.toJson(),
+      },
+    }));
+  }
+}
+
+class FakeRequest {
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/vmservice_test.dart:370]: I'm not sure about the naming here. FakeRequest doesn't say much, but FakeVmServiceRequest was just too verbose

File: packages/flutter_tools/test/general.shard/vmservice_test.dart [hunk 2]
```diff
@@ -280,8 +281,127 @@ void main() {
       ]))
     ]));
   });
+
+  testWithoutContext('runInView forwards arguments correctly', () async {
+    final FakeVmServiceHost fakeVmServiceHost = FakeVmServiceHost(
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/vmservice_test.dart:286]: Unlike the FakeProcessManager, this has to support both request/response and notifications. The latter have no corresponding request (like a log message, or isolate notification).

Combining these two concepts as a `VmServiceExpectation` allows us to describe  a more complex operation like runInView, which we need to ack the streamListen, ack the runInView request, and then return an isolate runnable notification

File: packages/flutter_tools/test/general.shard/vmservice_test.dart [hunk 3]
```diff
@@ -280,8 +281,127 @@ void main() {
...
+  });
+
+  final vm_service.Event event;
+  final String streamId;
+
+  @override
+  bool get isRequest => false;
+}
+
+
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/vmservice_test.dart:404]: Extra line

File: packages/flutter_tools/test/general.shard/vmservice_test.dart [hunk 4]
```diff
@@ -280,8 +281,127 @@ void main() {
...
+      }));
+    }
+  }
+}
+
+abstract class VmServiceExpectation {
+  bool get isRequest;
 }
 
+class FakeRequest implements VmServiceExpectation {
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/vmservice_test.dart:374]: > I'm not sure about the naming here. FakeRequest doesn't say much, but FakeVmServiceRequest was just too verbose

I like FakeVmServiceRequest better, it's not THAT long a name.

**Review by jmagman — APPROVED** [review]
LGTM

File: packages/flutter_tools/test/general.shard/vmservice_test.dart (hunk 4 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/vmservice_test.dart]: Done!

**Pull request merged**